File: Examples/Objective-C/Remote Control/Remote Control/FlashlightViewController.m
//
//  FlashlightViewController.m
//  Remote Control
//
//  Created by Calvin on 7/7/17.
//  Copyright © 2017 Bridgefy Inc. All rights reserved.
//

#import "FlashlightViewController.h"
#import "AdminViewTabBarController.h"

@interface FlashlightViewController ()

@property (weak, nonatomic) IBOutlet UIButton *flashlightButton;

@end

@implementation FlashlightViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)turnOnFlashlight:(id)sender {
    self.flashlightButton.enabled = NO;
    
    [self performSelector:@selector(enableFlashlightButton)
               withObject:nil
               afterDelay:15.0];
    
    if (self.tabBarController) {
        AdminViewTabBarController *tabBarController = (AdminViewTabBarController *)self.tabBarController;
        [tabBarController sendObject:nil sender:self];
    }
}

- (void)enableFlashlightButton {
    self.flashlightButton.enabled = YES;
}

@end

File: Examples/Swift/Remote Control/Remote Control/FlashlightViewController.swift
//
//  FlashlightViewController.swift
//  Remote Control
//
//  Created by Calvin on 7/11/17.
//  Copyright © 2017 Bridgefy Inc. All rights reserved.
//

import UIKit

class FlashlightViewController: UIViewController {
    
    @IBOutlet weak var flashlightButton: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func turnOnFlashlight(_ sender: Any) {
        self.flashlightButton.isEnabled = false
        
        self.perform(#selector(enableFlashlightButton),
                     with: nil,
                     afterDelay: 15)
        
        if (self.tabBarController != nil) {
            let tabBarController = self.tabBarController as! AdminViewTabBarController
            tabBarController.send(object: nil, sender: self)
        }
    }
    
    func enableFlashlightButton() {
        self.flashlightButton.isEnabled = true
    }

}
